### PR TITLE
[Numpy] Update QA Dataset and revise run_squad

### PR DESCRIPTION
**Pull request by ZheyuYe**
## Description ##
- [x] Downlaod script supports SearchQA, TriviaQA and HotpotQA
- [x] Update the squad results of Electra Model
- [x] revise the run_squad.py and implement the evaluation utils cloned from offical sciprt

## Comments ##
We are not going to provide the downlaod script for [Microsoft NewsQA](https://msropendata.com/datasets/939b1042-6402-4697-9c15-7a28de7e1321) and [Google Natural Questions](https://ai.google.com/research/NaturalQuestions/)

cc @dmlc/gluon-nlp-team

### CODE REVIEW

File: scripts/datasets/question_answering/README.md [hunk 1]
```diff
@@ -15,3 +16,62 @@ squad
...
+
+## TriviaQA
+[TriviaQA](https://nlp.cs.washington.edu/triviaqa/) is an open domain QA dataset. See more useful scripts in [Offical Github](https://github.com/mandarjoshi90/triviaqa)
+
+Run the following command to download triviaqa
+
+```bash
+python prepare_triviaqa.py --version rc         # Download TriviaQA version 1.0 for RC (2.5G)
+python prepare_triviaqa.py --version unfiltered # Download unfiltered TriviaQA version 1.0 (604M)
+```
```
Comment (sxjscience) [inline scripts/datasets/question_answering/README.md:38]: Besides mentioning that we are able to download via
```bash
python prepare_triviaqa.py --version rc         # Download TriviaQA version 1.0 for RC (2.5G)
python prepare_triviaqa.py --version unfiltered # Download unfiltered TriviaQA version 1.0 (604M)
```

We can also mention that we can download with
```bash
nlp_data prepare_triviaqa --version rc
nlp_data prepare_triviaqa --version unfiltered
```

File: scripts/question_answering/README.md [hunk 2]
```diff
@@ -93,23 +93,24 @@ For reference, we've included the results from Google's Original Experiments
 |ALBERT xlarge (googleresearch/albert)  | 92.9/86.4     | 87.9/84.1    |
 |ALBERT xxlarge (googleresearch/albert) | 94.6/89.1     | 89.8/86.9    |
 
-For BERT and ELECTRA model, the results on SQuAD1.1 and SQuAD2.0 are given as follows.
+For BERT and ELECTRA model, the results on SQuAD1.1 and SQuAD2.0 are given as follows. For ELECTRA model, same hyper-parameter as original paper are used, and layer-wise learning rate decay is applied (`layerwise_decay`) with the value of 0.8 for small/base and 0.9 for large.
```
Comment (sxjscience) [inline scripts/question_answering/README.md:96]: We can paste the command of finetuning Electra with layerwise lr decay. We may just show the command for the base model.

**Comment by szha** [comment]
CI needs fixing.

**Comment by sxjscience** [comment]
@szha  Seems that our test cases triggers some internal problems in MXNet.

**Comment by sxjscience** [comment]
@ZheyuYe Would you help upload this file to S3 and revise the link  https://github.com/dmlc/gluon-nlp/blob/85b6f0927c5417516e9daa26cde8256b169f33d8/src/gluonnlp/data/filtering.py#L195-L199? This is for language identification using fasttext.

File: src/gluonnlp/data/filtering.py [hunk 3]
```diff
@@ -186,14 +186,12 @@ def __init__(self, algo='fasttext_compressed', model_path=None):
...
-                    model_path = download('https://dl.fbaipublicfiles.com/fasttext/'
-                                          'supervised-models/lid.176.bin',
+                    model_path = download('s3://gluonnlp-numpy-data/misc/fasttext/lid.176.bin',
                                           os.path.join(get_model_zoo_home_dir(),
                                                        'fasttext_langid', 'lid.176.bin'),
                                           sha1_hash='e613bda316ecb4f5e1924140eedf81b81c087d9a')
                 elif algo == 'fasttext_compressed':
-                    model_path = download('https://dl.fbaipublicfiles.com/fasttext/'
-                                          'supervised-models/lid.176.ftz',
+                    model_path = download('s3://gluonnlp-numpy-data/misc/fasttext/lid.176.ftz',
```
Comment (sxjscience) [inline src/gluonnlp/data/filtering.py:194]: You may use `https://gluonnlp-numpy-data.s3-us-west-2.amazonaws.com/models/fasttext_langid/lid.176.ftz`

File: src/gluonnlp/data/filtering.py [hunk 4]
```diff
@@ -186,14 +186,12 @@ def __init__(self, algo='fasttext_compressed', model_path=None):
             fasttext = try_import_fasttext()
             if model_path is None:
                 if algo == 'fasttext':
-                    model_path = download('https://dl.fbaipublicfiles.com/fasttext/'
-                                          'supervised-models/lid.176.bin',
+                    model_path = download('s3://gluonnlp-numpy-data/misc/fasttext/lid.176.bin',
```
Comment (sxjscience) [inline src/gluonnlp/data/filtering.py:189]: You may use `https://gluonnlp-numpy-data.s3-us-west-2.amazonaws.com/models/fasttext_langid/lid.176.bin`

**Pull request merged**